StackAsLinkedList.java

```diff
@@ -0,0 +1,84 @@
+// Time Complexity : O(n)
+// Space Complexity : O(N)
+// Did this code successfully run on Leetcode :
+// Any problem you faced while coding this : N/A
+class StackNode {
+    int data;
+    //Creating a doubly linked list
+    StackNode next;
+    StackNode prev;
+    //Implementing it through doubly linkedList.
+
+    //Creating constructors
+    StackNode(){
+
+    }
+    StackNode(int data) {
+        //Constructor here
+        this.data = data;
+        this.next = null;
+        this.prev = null;
+    }
+    StackNode(int data, StackNode next) {
+        //Constructor here
+        this.data = data;
+        this.next = next;
+        this.prev = null;
+    }
+    StackNode(int data, StackNode next,StackNode prev) {
+        //Constructor here
+        this.data = data;
+        this.next = next;
+        this.prev = prev;
+    }
+}
+public class StackAsLinkedList {
+
+    StackNode root;
+
+    public boolean isEmpty() {
+        //Write your code here for the condition if stack is empty.
+        if(root == null){ // comparing whether the root is null or not
+            return true;
+        }
+        return false;
+    }
+
+    public void push(int data) {
+        //Write code to push data to the stack.
+        StackNode newNode = new StackNode(data); // creating node with input data
+        StackNode currNode = root; //creating other node to work on, so we don't lose the head node.
+        if (root == null) { // if root is null then put the given value in head node
+            root = newNode;
+        } else {
+            while (currNode.next != null) { // to traverse to the last node and then insert node there
+                currNode = currNode.next;
+            }
+            newNode.prev = currNode; // connecting the last and new node.
+            currNode.next = newNode;
+        }
+    }
+
+    public int pop() {

```

Comment:
You will be removing head node while pop().

Reply:
The thing is like you're implementing stack using LinkedList so all operations push(), pop(), peek() should be done in O(1) time right. So, that's why I'm telling that every node will be inserted at beginning of the linkedlist. So, when you pop() first node(which will be head node) will be deleted and same for peek(). For push also, insert node at head and make new inserted node as head.

Reply:
I think now you can imagine the stack in linked list.

Reply:
Thanks for the explanation. I implemented it in O(1) now. Please let me know if there is something which can be improved.